transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -201,7 +209,8 @@ private int epollWait(boolean oldWakenUp) throws IOException {
         long currentTimeNanos = System.nanoTime();
         long selectDeadLineNanos = currentTimeNanos + delayNanos(currentTimeNanos);
         for (;;) {
-            long timeoutMillis = (selectDeadLineNanos - currentTimeNanos + 500000L) / 1000000L;
+            long timeoutMillis = Math.min(MAX_EPOLL_WAIT_TIMEOUT,
```

Comment:
@normanmaurer @trustin - despite this PR being closed I would still appreciate if someone could provide clarity 👍